### PR TITLE
📈: Auto complete pull request if it's ready to merge

### PR DESCRIPTION
**Pull request by in-a-paddy**
## やったこと

- [x] Add WIP action to check if pull requests are work-in-progress
- [x] Add Auto-complete action to merge pull request if it got ready to merge

## やっていないこと

- Test if a pull request will be merged if title does not contain 'WIP' and someone approved

## 動作確認

- [x] Pull request approved by nobody was not merged
- [x] Pull request containing 'WIP' in title was not merged
- [x] Pull request containing 'WIP' in title was not merged even if someone approved
- [x] Auto-complete action runs if title of the pull request is changed
- [x] Cannot merge if "Restrict who can push to matching branches" protection is enabled

## デバイス

It's not about the devices.

## その他（レビュアーへの申し送り事項、懸念点など）

I can't think of anything.

### CODE REVIEW

**Review by Takumon — APPROVED** [review]
LGTM✨

**Review by Takumon — APPROVED** [review]
LGTM✨

File: .github/workflows/auto-complete.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  auto-complete:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+    steps:
+      - name: WIP
+        uses: wip/action@v1.0.0
+      - name: Auto-complete
+        if: ${{ success() }}
```
Comment (in-a-paddy) [inline .github/workflows/auto-complete.yaml:31]: This step takes too much time, so it is better to run it only when it is not a draft.

File: .github/workflows/auto-complete.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: Auto-complete
+
+on:
+  pull_request:
+    types:
+      - labeled
+      - unlabeled
+      - synchronize
+      - opened
```
Comment (in-a-paddy) [inline .github/workflows/auto-complete.yaml:9]: All pull requests require reviews, so can't merge just after opened

File: .github/workflows/auto-complete.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: Auto-complete
```
Comment (in-a-paddy) [inline .github/workflows/auto-complete.yaml:1]: test comment

File: .github/workflows/auto-complete.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+on:
+  pull_request_review:
+    types:
+      - submitted
+  check_suite:
+    types:
+      - completed
```
Comment (in-a-paddy) [inline .github/workflows/auto-complete.yaml:9]: Does not trigger...

File: .github/workflows/auto-complete.yaml (hunk 1 above)
Comment (in-a-paddy) [reply .github/workflows/auto-complete.yaml]: :ok:

**Review by Takumon — APPROVED** [review]
LGTM✨

**Pull request merged**